src/generic/datavgen.cpp

```diff
@@ -631,6 +633,101 @@ class wxDataViewTreeNode
 
 WX_DEFINE_SORTED_ARRAY_SIZE_T(unsigned int, wxDataViewSelection);
 
+
+struct RangeEntity
+{
+    unsigned int from;
+    unsigned int to;
+};
+
+WX_DECLARE_OBJARRAY(RangeEntity, ArrayOfRangeEntities);
+#include "wx/arrimpl.cpp"
+WX_DEFINE_OBJARRAY(ArrayOfRangeEntities)
+
+class RowRange
+{
+public:
+    void Add(const unsigned int idx);
+    void RemoveFrom(const unsigned int idx);
+    bool Has(const unsigned int idx);
```

Comment:
This method should be `const`, meaning that it should be written as

    bool Has(unsigned idx) const;

Notice that the `const` in the parameter declaration is unnecessary and is simply discarded by the C++ compiler, so it's better to remove it. However making the _method_ itself const is not unnecessary at all, as it prevents it from modifying the object and indicates to the callers that it doesn't do it.